internal/controller/pathwaysjob_controller.go

```diff
@@ -0,0 +1,663 @@
+	},
+}
+
+// Reconcile is part of the main kubernetes reconciliation loop which aims to
+// move the current state of the cluster closer to the desired state.
+
+// +kubebuilder:rbac:groups="",resources=events,verbs=create;watch;update;patch
+// +kubebuilder:rbac:groups=pathways-job.pathways.domain,resources=pathwaysjobs,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=pathways-job.pathways.domain,resources=pathwaysjobs/status,verbs=get;update;patch
+// +kubebuilder:rbac:groups=pathways-job.pathways.domain,resources=pathwaysjobs/finalizers,verbs=update
+// +kubebuilder:rbac:groups=jobset.x-k8s.io,resources=jobsets,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=jobset.x-k8s.io,resources=jobsets/status,verbs=get;update;patch
+func (r *PathwaysJobReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	pw := &pathwaysjob.PathwaysJob{}
+	log := ctrl.LoggerFrom(ctx).WithValues("pathwaysjob", klog.KObj(pw))
+	ctx = ctrl.LoggerInto(ctx, log)
+
+	log.Info("PathwaysJob: CONTROLLER WORKING...", "req.NamespacedName", req.NamespacedName.String(), "req.Namespace", req.Namespace)
+
+	// 1. Fetch the Pathways object
+	if err := r.Get(ctx, req.NamespacedName, pw); err != nil {
+		log.Info("PathwaysJob: Unable to fetch Pathways ")
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	// 2. Process the Pathways object and build a JobSet client
+	kubeconfig := ctrl.GetConfigOrDie()
+	// log.Info("PathwaysJob: config established...")
+
+	jobSetClient := jobsetclient.NewForConfigOrDie(kubeconfig)
+	// log.Info("PathwaysJob: client built for JobSet...")
+
+	// 2.1 Figure out if PathwaysJob is already present
+	// if it is the case, there is nothing to do.
+	// (ToDo) check states in "Suspended / Completed / Failed states",
+
+	childJobSet, err := r.getChildJobSet(ctx, pw, jobSetClient)
+	if err != nil {
+		log.Info("PathwaysJob: can't find JobSet, may create one!")
+		// return ctrl.Result{}, err
+	} else if childJobSet != nil {
+		log.Info("PathwaysJob: JobSet exists, not creating")
+		// 2.2 Find out JobSet's status
+		r.findJobSetStatus(ctx, childJobSet)
+		return ctrl.Result{}, nil
+	}
+
+	// 3. Update the cluster - create update and delete other resources
+	log.Info("PathwaysJob: creating JobSet \n")
+	if err := r.createJobSet(ctx, pw, jobSetClient); err != nil {
+		log.Error(err, "PathwaysJob: failed to create JobSet")
+		return ctrl.Result{}, err
+	}
+
+	//4. Update the object's status using Conditions (?)
+	childJobSet, _ = r.getChildJobSet(ctx, pw, jobSetClient)
+	r.findJobSetStatus(ctx, childJobSet)
+
+	//5. Return a result
+	log.Info("PathwaysJob: DONE DONE DONE!")
+	return ctrl.Result{}, nil
+}
+
+func (r *PathwaysJobReconciler) getChildJobSet(ctx context.Context, pw *pathwaysjob.PathwaysJob, jobSetClient *jobsetclient.Clientset) (*jobsetv1alpha2.JobSet, error) {
+	log3 := ctrl.LoggerFrom(ctx)
+	// .WithValues("pathwaysjob", klog.KObj(pw))
```

Comment:
Is this commented code still relevant? Same with the below

Reply:
It's a good to have, so added it to the logs. Thanks for the catch!